rag/advanced_rag/knowlege_compile/raptor.py

```
@@ -34,6 +36,325 @@
 
 from ._common import knowledge_compile_gen_conf
 
+# Claim extraction has its own concurrency budget, decoupled from the global
+# chat_limiter (which also serves clustering and embedding). MiniMax's API only
+# tolerates ~4 concurrent requests, so default to 4 and let it be overridden via
+# MAX_CONCURRENT_CLAIM_CHATS. On rate-limit errors each call backs off and retries
+# instead of letting the limiter itself scale up and hammer the API.
+_claim_limiter = LoopLocalSemaphore(int(os.environ.get("MAX_CONCURRENT_CLAIM_CHATS", 4)))
```

Ruff (0.16.3)

[warning] Invalid type for environment variable default; expected `str` or `None`

(PLW1508)